ui/src/pages/verify.tsx

```diff
@@ -0,0 +1,305 @@
+import { AxiosError } from 'axios';
+import Router from 'next/router';
+
+export const getServerSideProps: Types.GetServerSideProps = async (context) => {
+    Helpers.guardPrivateRoute(context);
+
+    let state: string | string[] | null = '/';
+    let newUser: boolean | null = false;
+
+    if (context.query.state) {
+        state = context.query.state;
+    }
+
+    if (context.query.newUser) {
+        newUser = Boolean(context.query.newUser == 'true');
+    }
+
+    return {
+        props: {
+            state,
+            newUser
+        }
+    };
+};
+
+type Props = {
+    state: string;
+    newUser: boolean;
+};
+
+const Verify: Types.NextPage<Props> = (props): React.ReactElement => {
+    const user = Stores.useAuthStore((state: Types.AuthStoreType) => state.user);
+    const setUser = Stores.useAuthStore((state) => state.setUser);
+    const setToast = Stores.useToastStore((state) => state.setToast);
+    const [email, setEmail] = React.useState('');
+    const [showCode, setShowCode] = React.useState(false);
+    const [code, setCode] = React.useState('');
+    const [loading, setLoading] = React.useState(false);
+    const [error, setError] = React.useState('');
+    const [success, setSuccess] = React.useState(false);
+
+    const submitEmail = () => {
+        setLoading(true);
+        requestCode();
+    };
+
+    const requestCode = async () => {
+        try {
+            setError('');
+            setCode('');
+
+            await api.get(`${Config.endpoints.verification}/${user?.orcid}?email=${email}`, user?.token);
+
+            setShowCode(true);
+            setLoading(false);
+
+            setToast({
+                visible: true,
+                dismiss: true,
+                title: 'Email sent',
+                icon: <OutlineIcons.MailIcon className="h-6 w-6 text-teal-400" aria-hidden="true" />,
+                message: 'Please check your inbox for your verification code.'
+            });
+        } catch (err) {
+            setLoading(false);
+            setToast({
+                visible: true,
+                dismiss: true,
+                title: 'Unable to send verification email',
+                icon: <OutlineIcons.MailIcon className="h-6 w-6 text-teal-400" aria-hidden="true" />,
+                message: 'Please check your email address and try again.'
+            });
+        }
+    };
+
+    const verifyCode = async () => {
+        try {
+            setError('');
+            setLoading(true);
+
+            // verify the supplied code, and if successful,
+            const getToken = await api.post<{ token: string }>(
+                `${Config.endpoints.verification}/${user?.orcid}`,
+                { code },
+                user?.token
+            );
+
+            // If success, decode the JWT, set the updated user, and redirect to state
+            if (getToken.status == 200) {
+                const decodedJWT = Helpers.setAndReturnJWT(getToken.data.token) as Types.UserType;
```

Comment:
Around this time from the video there is a visual bug where user is re added to the state and because user object X is different from user object Y at this point, so the nav is doubling up. React is rendering them both before routing to the new page where the state is normalised again. So i would guess to get around this, we would log the user out, &. re log them in again, and that process would re get the user

Reply:
Sure - I'll have a look at this. Not sure if logging out/in will be too jarring but definitely needs a fix :)

Reply:
Found this alternative fix:

https://github.com/JiscSD/octopus/pull/137/commits/8d2288ddcfc8cc2eb2d9cf0a07e0b1495a7015c6

Seems to do the trick, what do you think?

Reply:
👍🏼  bit of an odd one, but should be fine to be honest, especially if there is a navigation change.